cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -72,31 +79,40 @@ private String parse(String str, int depth) throws InvalidMacroException {
                                                     str, MAX_SUBSTITUTION_DEPTH));
     }
 
-    MacroMetadata macroPosition = findRightmostMacro(str);
+    MacroMetadata macroPosition = findRightmostMacro(str, str.length());
```

Comment:
hmm, can you add some comment/clarification, the api documentation for `findRightmostMacro` says 2nd param is the position to start the search for macro from, if thats the string length, this will always be empty or null right ?

Reply:
ok i can clarify it in the comments. It's the position to start searching backwards from, similar to String's .lastIndexOf() method.